planner/core/stats.go

```diff
@@ -509,7 +509,8 @@ func (ts *LogicalTableScan) DeriveStats(_ []*property.StatsInfo, _ *expression.S
 	// ts.Handle could be nil if PK is Handle, and PK column has been pruned.
 	// TODO: support clustered index.
 	if ts.HandleCols != nil {
-		ts.Ranges, err = ranger.BuildTableRange(ts.AccessConds, ts.ctx, ts.HandleCols.GetCol(0).RetType)
+		// ts.AccessConds have already been determined here so we can build ranges without the memory limit.
```

Comment:
If the `ts.AccessConds have already been determined`, it means we have already build the range before? Why not store it and just use it?

Reply:
+1

Reply:
I made a mistake. The ranges have not been built before. `(*LogicalTableScan).DeriveStats` is only used for cascades optimizer and cascades optimizer seems to separate `DetachCondsForColumn` and `BuildTableRange` apart. `DetachCondsForColumn` is in `(*PushSelDownTableScan).OnTransform` and `BuildTableRange` is in `(*LogicalTableScan).DeriveStats`. I'm not very sure how to give `remainedConds` returned by `BuildTableRange` back to `LogicalSelection` upon `LogicalTableScan`. I don't restrict range mem usage here and add TODO.